### PR TITLE
Fix DAC alerts and errors

### LINKED ISSUE #8721: Connecting to DAC locally Raises alerts and puts error in log
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

When I run the below command locally, from the server, (sql agent job or Powershell ISE.). Only locally.  My process connects and works, it just some how gets two connections.  I feel like it has something to do with the DAC listening on the Loop back address and also running it locally but not sure what,

$dac = Connect-DbaInstance -SqlInstance $PrimaryInstance -DedicatedAdminConnection;

Date		1/12/2023 1:34:09 AM
Log		SQL Server (Current - 1/12/2023 9:28:00 AM)

Source		Logon

Message
Could not connect because the maximum number of '1' dedicated administrator connections already exists. Before a new connection can be made, the existing dedicated administrator connection must be dropped, either by logging off or ending the process. [CLIENT: 127.0.0.1]

Date		1/12/2023 1:34:09 AM
Log		SQL Server (Current - 1/12/2023 9:28:00 AM)

Source		Logon

Message
Error: 17810, Severity: 20, State: 1.



### Steps to Reproduce

```powershell
# provide your command(s) executed pertaining to dbatools
# please include variable values (redacted or fake if needed) for reference
```


### Please confirm that you are running the most recent version of dbatools

Run $dac = Connect-DbaInstance -SqlInstance $PrimaryInstance -DedicatedAdminConnection;
on the same server as SQL Server instance is installed.

### Other details or mentions

_No response_

### What PowerShell host was used when producing this error

Windows PowerShell ISE (powershell_ise.exe)

### PowerShell Host Version


Name                           Value                                                                                                                                                                                                                                                                                                                                        
----                           -----                                                                                                                                                                                                                                                                                                                                        
PSVersion                      5.1.17763.2931                                                                                                                                                                                                                                                                                                                               
PSEdition                      Desktop                                                                                                                                                                                                                                                                                                                                      
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                                                                                                                                                      
BuildVersion                   10.0.17763.2931                                                                                                                                                                                                                                                                                                                              
CLRVersion                     4.0.30319.42000                                                                                                                                                                                                                                                                                                                              
WSManStackVersion              3.0                                                                                                                                                                                                                                                                                                                                          
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                                                                                                                                                          
SerializationVersion           1.1.0.1                                                                                                                                                                                                                                                                                                                                      




### SQL Server Edition and Build number

Microsoft SQL Server 2019 (RTM-CU16-GDR) (KB5014353) - 15.0.4236.7 (X64)   May 29 2022 15:55:47   Copyright (C) 2019 Microsoft Corporation  Enterprise Edition: Core-based Licensing (64-bit) on Windows Server 2019 Standard 10.0 <X64> (Build 17763: ) 

### .NET Framework Version

PSChildName                      Version       
-----------                      -------       
v2.0.50727                       2.0.50727.4927
v3.0                             3.0.30729.4926
Windows Communication Foundation 3.0.4506.4926 
Windows Presentation Foundation  3.0.6920.4902 
v3.5                             3.5.30729.4926
Client                           4.7.03190     
Full                             4.7.03190     
Client                           4.0.0.0

### PR DESCRIPTION
**Pull request by potatoqualitee**
basically, we gotta skip a few of the checks and return the object earlier. we will not be able to show the user/Connected As because getting that info causes an alert/error in the log.

fixes #8721

### CODE REVIEW

**Comment by potatoqualitee** [comment]
omg there are not 918 changes, its bc i built it off of another branch with that many changes. someone please help by reviewing all the PRs so i can merge 💀

File: public/Connect-DbaInstance.ps1 [hunk 1]
```diff
@@ -917,16 +919,18 @@ function Connect-DbaInstance {
...
-            Write-Message -Level Debug -Message "We connect to the instance by running SELECT 'dbatools is opening a new connection'"
             try {
-                $null = $server.ConnectionContext.ExecuteWithResults("SELECT 'dbatools is opening a new connection'")
+                if (-not $DedicatedAdminConnection) {
+                    Write-Message -Level Debug -Message "We connect to the instance by running SELECT 'dbatools is opening a new connection'"
+                    $null = $server.ConnectionContext.ExecuteWithResults("SELECT 'dbatools is opening a new connection'")
+                    Write-Message -Level Debug -Message "We have a connected server object"
+                }
             } catch {
                 if ($DedicatedAdminConnection) {
```
Comment (andreasjordan) [inline public/Connect-DbaInstance.ps1:929]: We don't need this if we don't execute the query for a DAC.

Comment (potatoqualitee) [reply public/Connect-DbaInstance.ps1:929]: true, ty! ill clean that up

**Comment by andreasjordan** [comment]
Code looks good to me.

**Comment by S0l4r15** [comment]
I know this is quite old now but we still get : 
Could not connect because the maximum number of '1' dedicated administrator connections already exists. Before a new connection can be made, the existing dedicated administrator connection must be dropped, either by logging off or ending the process. [CLIENT: 127.0.0.1]

This is using the a version we installed in Nov 2023.  We're "syncing" logins, jobs, linked servers ... pretty much everything apart from the databases to a secondary node in an AAG on the hour.  Every hour we get 3 of those alerts which I understand are generated due to a need for DAC.

Should this now be fixed?
If not, can the process which generates the DAC connections be done in serial (i.e. wait for a disconnect before establishing a new DAC)?

Kind regards,

Carl

**Comment by andreasjordan** [comment]
You could open a new issue for that. But: I would need to be able to reproduce the problem in my lab to be able to provide a better solution. So try to provide information about the needed components to reproduce the problem.

**Comment by 00-schaudhary** [comment]
@andreasjordan - I have posted the below comment in #9204.

I have the same issue. 

Connect-DbaInstance -SqlInstance localhost\instanceName -DedicatedAdminConnection -verbose

This generates the error in the SQL error logs if I use the name (localhost or serverName).
No error is generated if an IP is used (127.0.0.1 or serverIP).

I have a Windows Cluster and using Availability Groups and AG listeners. So multiple IPs are in use. I think for some reason it tries to establish a connection on multiple IPs, when using the name.

Hope this helps is recreating the issue internally.

**Pull request merged**